### PR TITLE
859 Use FlowSimulation in Python Bindings

### PR DESCRIPTION
**Pull request by HenrZu**
# Changes and Information

Please **briefly list the changes** made, additional Information and what the Reviewer should look out for:

-

## Merge Request - Guideline Checklist

Please check our [git workflow](https://github.com/SciCompMod/memilio/wiki/git-workflow). Use the **draft** feature if the Pull Request is not yet ready to review.

### Checks by code author

- [x] Every addressed issue is linked (use the "Closes #ISSUE" keyword below)
- [x] New code adheres to [coding guidelines](https://github.com/SciCompMod/memilio/wiki/Coding-guidelines)
- [x] No large data files have been added (files should in sum not exceed 100 KB, avoid PDFs, Word docs, etc.)
- [x] Tests are added for new functionality and a local test run was successful
- [x] Appropriate **documentation** for new functionality has been added (Doxygen in the code and Markdown files if necessary)
- [x] Proper attention to licenses, especially no new third-party software with conflicting license has been added
- [ ] (For ABM development) Checked [benchmark results](https://github.com/SciCompMod/memilio/wiki/Agent-Based-Model-Development)

### Checks by code reviewer(s)

- [x] Corresponding issue(s) is/are linked and addressed
- [x] Code is clean of development artifacts (no deactivated or commented code lines, no debugging printouts, etc.)
- [x] Appropriate **unit tests** have been added, CI passes and code coverage is acceptable (did not decrease)
- [x] No large data files added in the whole history of commits(files should in sum not exceed 100 KB, avoid PDFs, Word docs, etc.)

### CODE REVIEW

File: pycode/examples/simulation/ode_seir_flows.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    model.parameters.ContactPatterns.baseline = np.ones((1, 1))
+    model.parameters.ContactPatterns.minimum = np.zeros((1, 1))
+    model.parameters.ContactPatterns.add_damping(
+        Damping(coeffs=np.r_[0.9], t=30.0, level=0, type=0))
+
+    # Check logical constraints to parameters
+    model.check_constraints()
+
+    # Run flow simulation
+    seir = simulate_flows(0, days, dt, model)
```
Comment (MaxBetzDLR) [inline pycode/examples/simulation/ode_seir_flows.py:69]: Speaking variable names and more pythonic
```suggestion
    (result, flows) = simulate_flows(0, days, dt, model)
```

File: pycode/examples/simulation/ode_seir_flows.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    print(seir[0].print_table(["S", "E", "I", "R"], 16, 5))
```
Comment (MaxBetzDLR) [inline pycode/examples/simulation/ode_seir_flows.py:71]:
```suggestion
    print(result.print_table(["S", "E", "I", "R"], 16, 5))
```

File: pycode/examples/simulation/ode_seir_flows.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    print(seir[1].print_table(["S->E", "E->I", "I->R"], 16, 5))
```
Comment (MaxBetzDLR) [inline pycode/examples/simulation/ode_seir_flows.py:72]:
```suggestion
    print(flows.print_table(["S->E", "E->I", "I->R"], 16, 5))
```

File: pycode/examples/simulation/ode_seir_flows.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+
+if __name__ == "__main__":
+    arg_parser = argparse.ArgumentParser(
+        'ode seir model with flow simulation',
+        description='Simple example demonstrating the setup and simulation of the OSEIR model while using the flow simulation.')
```
Comment (MaxBetzDLR) [inline pycode/examples/simulation/ode_seir_flows.py:78]:
```suggestion
        description='Simple example demonstrating the setup and flow simulation of the OSEIR model.')
```

**Review by MaxBetzDLR — CHANGES_REQUESTED** [review]

**Comment by MaxBetzDLR** [comment]
Is there a reason, not to add a flow simulation to the osir model aswell?

**Comment by HenrZu** [comment]
Thank you for the review. By merging the #511 PR, we have replaced the model equations in all models (except the SIR) with the flow notation. However, since we continue to support the old notation of the models, the SIR model now serves as a demonstration that it is possible :)

**Pull request merged**